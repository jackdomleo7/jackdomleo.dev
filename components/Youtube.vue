<template>
  <iframe class="youtube" width="560" height="315" :title="`${videoTitle} on YouTube`" :src="`https://www.youtube.com/embed/${urlHash}`" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'Youtube',
  props: {
    urlHash: {
      type: String,
      required: true
    },
    videoTitle: {
      type: String,
      required: true
    }
  }
})
</script>

<style lang="scss" scoped>
.youtube {
  min-height: 25rem;
}
</style>