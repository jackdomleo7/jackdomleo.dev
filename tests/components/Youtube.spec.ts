import { mount, shallowMount } from '@vue/test-utils'
import { configureAxe, toHaveNoViolations } from 'jest-axe'
import Youtube from '@/components/Youtube.vue'

expect.extend(toHaveNoViolations)

const axe = configureAxe({
  rules: {
    region: { enabled: false } // Don't need to check because components are loaded in isolation
  }
})

describe('Youtube', () => {
  /**
   * Vue instance
   */

  it('is a Vue instance', () => {
    const wrapper = mount(Youtube, {
      propsData: {
        urlHash: 'wKbU8B-QVZk',
        videoTitle: "I Can't Believe You've Done This"
      }
    })

    expect(wrapper.vm).toBeTruthy()
  })

  /**
   * Snapshots
   */

  it('matches snapshot when mounted', () => {
    const wrapper = mount(Youtube, {
      propsData: {
        urlHash: 'wKbU8B-QVZk',
        videoTitle: "I Can't Believe You've Done This"
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('matches snapshot when shallow mounted', () => {
    const wrapper = shallowMount(Youtube, {
      propsData: {
        urlHash: 'wKbU8B-QVZk',
        videoTitle: "I Can't Believe You've Done This"
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  /**
   * A11y
   */

  it('to have no obvious accessibility violations', async () => {
    const wrapper = mount(Youtube, {
      propsData: {
        urlHash: 'wKbU8B-QVZk',
        videoTitle: "I Can't Believe You've Done This"
      }
    })

    await expect(axe(wrapper.element)).resolves.toHaveNoViolations()
  })
})
