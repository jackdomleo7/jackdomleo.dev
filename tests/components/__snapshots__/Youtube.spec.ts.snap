// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Youtube matches snapshot when mounted 1`] = `
<iframe
  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
  allowfullscreen="allowfullscreen"
  class="youtube"
  frameborder="0"
  height="315"
  src="https://www.youtube.com/embed/wKbU8B-QVZk"
  title="I Can't Believe You've Done This on YouTube"
  width="560"
/>
`;

exports[`Youtube matches snapshot when shallow mounted 1`] = `
<iframe
  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
  allowfullscreen="allowfullscreen"
  class="youtube"
  frameborder="0"
  height="315"
  src="https://www.youtube.com/embed/wKbU8B-QVZk"
  title="I Can't Believe You've Done This on YouTube"
  width="560"
/>
`;
